### DISCUSSION TITLE
Question about vSelect

### DISCUSSION THREAD

**Discussion by PsychoFox11** [sanni/cartreader#811] [Need help]
Looking to get a Sanni with the vSelect (voltage switch). However, with the manual voltage switch, I always set it before power-on, as I've noticed carts get power right away (due to some carts with LEDs in them which light up). This ensures I give all carts the right voltage.

With the automatic one, when I change from, for example, dumping games for a 5v console to ones for a 3v console (or vice-versa), what does it feed the cart on power-on? Does it detect something and use the right voltage, use the last used voltage, or always the same default? Do I need to wait until I've selected the console to insert the cart?

**Comment by cmcm711** [comment]
vSelect models supply 3.3v on power-on, then switch to 5v when a game system that requires it is selected. This way, it won't feed 5v to a 3.3v cartridge.

**Reply by PsychoFox11** [reply]
I'm not an electrical engineer obviously haha, but I'd heard that low voltage is bad for things as well, does it have any potential negative effects on 5v stuff (which is most slots/adapters)?

**Reply by KSOFM** [reply]
If you're making/getting an OSCR for backing up save files, this is when you should really be concerned as seems to effect the voltage that the SRAM gets which in turn effects your save file. As I've found, due to a dodgy switch that didn't quite slide fully across to change the voltage to 5V, it corrupted the save file on two of my GB games.

Both the save on the cartridge and the save file it downloaded to the SD card were unrecoverable which was really upsetting as the tiny CR1616 battery in my Pokemon cart had kept that save going for 25 years until the OSCR depleted it. However, new battery holder/battery installed and they both work fine.

I'm not sure exactly why this happened as the battery itself is 3v but I'm not going to investigate it further, I can only assume under powering the cartridge lead to the voltage on the SRAM to dip too low...

As people have said bellow, the ROM chip is pretty tough and unlikely to be damaged if operated slightly over or under voltage for a short amount of time.

The power switch on my OSCR is just as dodgy, I will be replacing both when I add the shell/enclosure with decent switches and I'll be adding a couple of LEDs as voltage selection indicators.

**Reply by Ancyker** [reply]
I've not had this issue and the games I actually use with the OSCR (as in games I play that aren't just for testing it) are mostly Pokemon games (Red, Yellow, Crystal, and Ruby primarily). Because of that, I doubt it's simply because you had the switch set to 3.3V instead of 5V (if you think about it, the battery is 3V, 3V is what the SRAM chip wants).

If by "dodgy switch" you mean one that is intermittently disconnecting the power or quickly changing the voltage, that can mess with things. It's similar to the primary exploit method used by many mod chips and security exploits called "voltage glitching" where the attacker is making rapid and repeated voltage changes, which may include removing voltage entirely for a brief moment (microseconds). They do this to try to make the target chip glitch and do something unintended, such as starting without some sort of protection enabled. 

Doing this to a cart with SRAM can definitely ruin the save. It'd be the same as rapidly turning a Game Boy on and off which is known to corrupt save files and people have had systems (GBAs especially) with faulty power buttons corrupt their saves due to this. The power switches we use aren't normally bad, however, it's possible yours was overheated and damaged during soldering. It's also possible you got some from a bad batch. But if your power or voltage switch is flaky you should replace it.

With that said, for reasons that have nothing to do with this, [the new revision](https://github.com/sanni/cartreader/discussions/874#discussioncomment-9495622) does have an option to toggle the power to the cartridge slots independently of the rest of the hardware. This should absolutely solve any issue related to the cartridge and when/how voltage and power are applied. However, that feature requires installing the ATmega2560 and other components directly onto the main PCB which is fairly difficult and more expensive when making just 1.

**Reply by KSOFM** [reply]
Yeah, I think I was jumping to conclusions because I was upset and trying to think of what could have gone wrong. I've now realised it also lost my Secret of Mana (SNES) save which had a save slot from a childhood friend who is longer with us :(

Having spent more time with it, I've noticed especially for the N64 connector that the pins are the wrong pitch and often don't line up very well. That could also have been the cause. I did carefully clean all connectors on my games before trying to read them. I kept re-seating until I good read of the ROM where it passed the CRC check before reading the save but at some point during that process, the save has been lost. Typically, I didn't really care about most other save files and lost the ones I wanted most.

The GB slot seems to be one of the more reliable. The SNES slot was about 50/50 and the N64 slot was terrible. Having said that, I didn't lose anything from any N64 games.

The main thing is I've tested cross compatibility with the Analogue Pocket, SD2SNES and Everdrive 64 which all worked flawlessly and I feel confident uploading saves back to cartridges after replacing batteries in future.

**Reply by Ancyker** [reply]
> Having spent more time with it, I've noticed especially for the N64 connector that the pins are the wrong pitch and often don't line up very well.

You bought it from Aliexpress or the person shipping from China on eBay?

**Reply by Ancyker** [reply]
@KSOFM If so, see: https://github.com/sanni/cartreader/discussions/835#discussioncomment-6770793

**Comment by PsychoFox11** [comment]
Still curious, is there any chance of damage due to undervoltage? I remember the discussion with older everdrives running at 3v potentially harming 5v consoles (I'd expect the concern to be with the console harming the cart, but it seemed to be the other way around. I don't think there was any reports of it happening, but many said science-wise it was possible.). Just trying to preserve my carts so definitely don't wanna do anything risky.

**Reply by Ancyker** [reply]
This isn't really something that can be absolutely answered by anyone here as most of the chips used were either proprietary or never had their data sheets uploaded to the Internet. If we look at a 5V chip used during that era that is common in DIY flash carts (as well as the official N64 dev carts), we can see the safe voltage ranges under Absolute Maximum Ratings:
![image](https://github.com/sanni/cartreader/assets/7105333/1883f683-bd1b-47c7-a158-2fc4ee094dc9)

These are the "safe" ranges that won't normally damage the device. This is different from the operating ranges, which for the same chip are:
![image](https://github.com/sanni/cartreader/assets/7105333/6d520eb2-6bd7-497a-a521-bb810a18657b)

The operating ranges are what the chip will function normally under, going outside of those will generally result in instability or other undesired behavior. Going outside the absolute maximum ratings is where damage occurs.

This is a different chip than is in most carts, but it gives insight into what was typical at the time. Generally speaking, lower voltages (as long as they aren't negative) won't hurt things, but there are almost always exceptions to any rule. As long as you aren't leaving a cart in a powered reader for long periods of time it should be fine. That's bad for the cart anyway, even if you were using a manual switch.

**Reply by Ancyker** [reply]
Also, regarding the damage vs 5V/3.3V, you got it backward. A 5V system will damage a 3.3V cartridge. This is because cheap flash cartridges would often buck the 5V down to 3.3V for VCC, but not translate the IO to 3.3V, driving the 3.3V IO pins at 5V. That's outside the absolute maximum rating of most 3.3V chips (which tend to top out around 3.5-4.5V). On top of that, overdriving is worse than underdriving, so this is especially bad.

In regards to the chip in my previous reply, it will register anything over 2.0V as a logic high, while anything under 0.8V is a logic low. That means even if we tried to talk to it with 3.3V we'd be fine, but it talking back would be at 5V which could be bad if we were talking to it with a 3.3V device that was not 5V tolerant.

And if that wasn't enough, the use case with flash carts is also quite different: In the case of the OSCR, you are inserting a cart for a few minutes to dump it. In the case of a flash cart, you are connecting to a system and using it for hours upon hours. That's an important consideration, as the length of time spent outside normal operating conditions does affect the lifespan of any device -- especially while using that device.
